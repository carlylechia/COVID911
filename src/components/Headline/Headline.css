* {
  box-sizing: border-box;
}

.block {
  display: block;
  margin: auto;
}

.none {
  display: none;
}

#headline {
  background: #d70040;
  display: block;
}

.structure {
  display: flex;
  justify-content: center;
  align-items: flex-start;
}

.topic {
  color: #fff;
  text-align: start;
  width: 40%;
  word-break: break-all;
  padding: 5rem 1rem;
}

.maps {
  width: 60%;
  height: 30vh;
  object-fit: contain;
}

@media (max-width: 600px) {
  .structure {
    display: block;
  }

  .topic {
    color: aliceblue;
    text-align: center;
    min-width: 80%;
    max-width: 100%;
    word-break: normal;
    padding: 1.5rem 0;
    margin: 0.5rem auto;
  }

  .maps {
    width: 90%;
    height: auto;
    margin: 1rem auto;
  }
}
