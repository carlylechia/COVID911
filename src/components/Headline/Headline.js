import { useLocation } from 'react-router-dom';
import { useSelector, shallowEqual } from 'react-redux';
import Search from '../Search/Search';
import './Headline.css';
import CountryImage from '../Countries/Country/CountryImage';

const Headline = () => {
  const { pathname } = useLocation();
  const mapCondition = pathname.replace('/', '');
  const newCases = useSelector((state) => state.regionsReducer.newCases, shallowEqual);
  const countryName = mapCondition.replace(/[_]/g, '-').replace('*', '').replace(',', '').toUpperCase();
  return (
    <div id="headline" className="flex ma0 justify-center items-center relative">
      <div className="structure">
        <CountryImage
          className={`${!mapCondition.length ? 'block' : 'none'} o-30 w-auto-ns w-50 h-100-ns h-50 maps`}
          alt=""
          src="world-map"
        />
        <CountryImage
          className={`${mapCondition.length ? 'block' : 'none'} o-30 w-auto-ns w-50 h-100-ns h-50 maps`}
          alt=""
          src={mapCondition}
        />

        <h1 className="topic white f2-l f4-m f4-ns f5">
          {!mapCondition.length ? (
            <span style={{ whiteSpace: 'pre-line' }}>
              {'Covid Statistics<br/> around the world.'.split('<br/>').join('\n')}
            </span>
          )
            : (
              <>
                <p>
                  {countryName}
                </p>
                <p>
                  Confirmed New Cases:
                  {newCases}
                </p>
              </>
            )}
        </h1>
      </div>
      <Search />
    </div>
  );
};
export default Headline;
