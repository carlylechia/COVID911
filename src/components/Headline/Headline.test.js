import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../../Redux/createStore';
import Headline from './Headline';

describe('Headline component test', () => {
  test('Headline snapshot', () => {
    const headline = render(
      <Router>
        <Provider store={store}>
          <Headline />
        </Provider>
      </Router>,
    );
    expect(headline).toMatchSnapshot();
  });
});
