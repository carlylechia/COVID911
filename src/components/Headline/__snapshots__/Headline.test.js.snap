// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Headline component test Headline snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex ma0 justify-center items-center relative"
        id="headline"
      >
        <div
          class="structure"
        >
          <img
            alt=""
            class="block o-30 w-auto-ns w-50 h-100-ns h-50 maps"
            src="/images/world-map.png"
          />
          <img
            alt=""
            class="none o-30 w-auto-ns w-50 h-100-ns h-50 maps"
            src="/images/.png"
          />
          <h1
            class="topic white f2-l f4-m f4-ns f5"
          >
            <span
              style="white-space: pre-line;"
            >
              Covid Statistics
 around the world.
            </span>
          </h1>
        </div>
        <input
          class="block"
          placeholder="Search Country"
          style="border-width: 2px; border-radius: 1rem; padding: 5px; font-size: 1rem; text-align: center; margin: 2rem auto;"
          type="text"
          value=""
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex ma0 justify-center items-center relative"
      id="headline"
    >
      <div
        class="structure"
      >
        <img
          alt=""
          class="block o-30 w-auto-ns w-50 h-100-ns h-50 maps"
          src="/images/world-map.png"
        />
        <img
          alt=""
          class="none o-30 w-auto-ns w-50 h-100-ns h-50 maps"
          src="/images/.png"
        />
        <h1
          class="topic white f2-l f4-m f4-ns f5"
        >
          <span
            style="white-space: pre-line;"
          >
            Covid Statistics
 around the world.
          </span>
        </h1>
      </div>
      <input
        class="block"
        placeholder="Search Country"
        style="border-width: 2px; border-radius: 1rem; padding: 5px; font-size: 1rem; text-align: center; margin: 2rem auto;"
        type="text"
        value=""
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
